syserol/impute.py

```diff
@@ -43,7 +43,7 @@ def evaluate_missing(comps, numSample=15, chords=True, PCAcompare=False):
         if chords:
             missingCube[:, j, k] = np.nan
         else:
-            while sum(np.isnan(missingCube[:, j, k])) > len(missingCube[:, j, k]) - 2:
+            while sum(np.isnan(missingCube[:, j, k])) > len(missingCube[:, j, k]) - 10:
```

Comment:
This check is wrong. If we end up with a chord having less than 10 values, this just doesn't remove as many values as intended. Instead, it should start the whole process of building missingCube over again.

Better would be if it checked whether the element chosen by random.choice falls into a chord with 10 values, and picked a new position if so.

Reply:
I thought this is what it was doing, or at least that is what I intended? It's a while loop that will continue picking positions until it finds a position that is not in a chord with less than 10 values. Then it will make that value nan.

Reply:
Oh clever. You're right.